src/components/Articles.jsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom";
+import { getAllArticles } from "../utils/api";
+import ArticleCard from "./ArticleCard";
+
+const TopicArticles = () => {
+    const {topic} = useParams();
+    const [isLoading, setIsLoading] = useState(true);
+    const[articlesByTopic, setArticlesByTopic] = useState([]);
+    const [params, setParams] = useState({})
+
```

Comment:
You might find it useful to look up useSearchParams as an alternative way of handling form inputs (this is optional and up to you though!)